cmd/dump/src/lib.rs

```diff
@@ -48,34 +48,895 @@
 //! ```
 //!
 
-use anyhow::Result;
+use anyhow::{anyhow, bail, Result};
 use clap::Command as ClapCommand;
-use clap::{CommandFactory, Parser};
+use clap::{ArgGroup, CommandFactory, Parser};
+use core::mem::size_of;
+use goblin::elf::Elf;
+use hif::*;
+use humility::arch::ARMRegister;
 use humility::cli::Subcommand;
+use humility::core::Core;
+use humility::hubris::*;
+use humility_cmd::hiffy::*;
+use humility_cmd::idol::{self, HubrisIdol};
 use humility_cmd::{Archive, Attach, Command, Validate};
+use humpty::{
+    DumpAreaHeader, DumpRegister, DumpSegmentData, DumpSegmentHeader,
+};
+use indicatif::{HumanBytes, HumanDuration};
+use indicatif::{ProgressBar, ProgressStyle};
+use num_traits::FromPrimitive;
+use std::cell::RefCell;
+use std::collections::BTreeMap;
+use std::collections::HashMap;
+use std::io::Cursor;
+use std::io::Read;
+use std::path::Path;
+use std::time::{Duration, Instant};
+use zerocopy::FromBytes;
 
 #[derive(Parser, Debug)]
-#[clap(name = "dump", about = env!("CARGO_PKG_DESCRIPTION"))]
+#[clap(
+    name = "dump", about = env!("CARGO_PKG_DESCRIPTION"),
+    group = ArgGroup::new("simulation").multiple(false)
+        .required(false).requires("force-dump-agent")
+)]
+
 struct DumpArgs {
+    /// sets timeout
+    #[clap(
+        long, short = 'T', default_value_t = 5000, value_name = "timeout_ms",
+        parse(try_from_str = parse_int::parse)
+    )]
+    timeout: u32,
+
+    /// show dump agent status
+    #[clap(long, conflicts_with = "simulation")]
+    dump_agent_status: bool,
+
+    /// force use of the dump agent when directly attached with a debug probe
+    #[clap(long)]
+    force_dump_agent: bool,
+
+    /// force existing in situ dump to be read
+    #[clap(long, conflicts_with = "simulation")]
+    force_read: bool,
+
+    /// initialize dump state, clearing any dump at the dump agent
+    #[clap(long, conflicts_with = "simulation")]
+    initialize_dump_agent: bool,
+
+    /// overwrite any dump state as part of taking dump
+    #[clap(
+        long, short = 'F',
+        conflicts_with_all = &["simulation", "initialize-dump-agent"]
```

Comment:
I think this should only conflict with `simulate_dumper`, not the entire `simulation` group; right now I see the following

```console
➜  humility git:(dumper) ✗ cargo run -- dump --emulate-dumper --force-dump-agent
    Finished dev [unoptimized + debuginfo] target(s) in 0.49s
     Running `target/debug/humility dump --emulate-dumper --force-dump-agent`
humility: attached via ST-Link V3
humility dump failed: there appears to already be a dump in situ; clear with --initialize-dump-agent or force dump to be overwritten with --force-overwrite
➜  humility git:(dumper) ✗ cargo run -- dump --emulate-dumper --force-dump-agent --force-overwrite
    Finished dev [unoptimized + debuginfo] target(s) in 0.36s
     Running `target/debug/humility dump --emulate-dumper --force-dump-agent --force-overwrite`
error: The argument '--force-overwrite' cannot be used with:
    --simulate-dumper
    --emulate-dumper

USAGE:
    humility dump --emulate-dumper --force-dump-agent --force-overwrite

For more information try --help
```

(this may be true of `--initialize-dump-agent` as well)

Reply:
Ha, yes -- had just independently made this change as part of testing your Nucleo fix. ;)